### PR TITLE
[clipper2] Add new port

### LINKED ISSUE #26448: [New Port Request] Clipper2
Library name: Clipper2

Library description:
A Polygon [Clipping](https://en.wikipedia.org/wiki/Clipping_(computer_graphics)) and [Offsetting](https://en.wikipedia.org/wiki/Parallel_curve) library (in C++, C# & Delphi)
Clipper2 is a major update of my original [Clipper](https://sourceforge.net/projects/polyclipping/) library which I'm now calling Clipper1.
Clipper1 was written over 10 years ago and still works very well, but Clipper2 is better in just about every way.
Source repository URL:

Project homepage (if different from the source repository):
https://github.com/AngusJohnson/Clipper2

### PR DESCRIPTION
**Pull request by ex-purple**
Fixes https://github.com/microsoft/vcpkg/issues/26448

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by dg0yt — COMMENTED** [review]
All possible file conflicts with clipper seem to be resolved by upstream using `2` in file and directory names.
I don't know about symbol names.

File: ports/clipper2/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "clipper2",
```
Comment (Adela0814) [inline ports/clipper2/vcpkg.json:2]: Please change the port name to clipper2, thanks.
A name must be an identifier: i.e., it must only consist of lowercase ascii alphabetic characters, numbers, and hyphens, and it must not begin nor end with a hyphen. [Link](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#:~:text=A%20name%20must%20be%20an%20identifier%3A%20i.e.%2C%20it%20must%20only%20consist%20of%20lowercase%20ascii%20alphabetic%20characters%2C%20numbers%2C%20and%20hyphens%2C%20and%20it%20must%20not%20begin%20nor%20end%20with%20a%20hyphen.).

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/clipper2/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/clipper2/vcpkg.json:2]: The name is `clipper2`, isn't it?

Comment (Adela0814) [reply ports/clipper2/vcpkg.json:2]: Confused, I just saw it was `Clipper 2`. Ok, ignore this.

**Comment by Adela0814** [comment]
I got a bad hash error in my local:
```
error: Failed to download from mirror set
error: File does not have the expected hash:
url: https://github.com/AngusJohnson/Clipper2/archive/Clipper2_1.1.1.tar.gz
File: E:\3\vcpkg\downloads\AngusJohnson-Clipper2-Clipper2_1.1.1.tar.gz.936.part
Expected hash: b2b35cf4d03a387f43ee65ea49a30ec194d75ecf1b9b7431263c9073ee35ac63419ae3714f15220e5865e437b2bb5d9863cc01c5a3844304bb61933ae8c03c5b
Actual hash: e996ef8a2ec412189f0ba95a6f200c0818f9755930b05cd20b630d33760dec619c6a735ac056f5dfbaccf793bf6ebeb7b6c102fd9ff83b0d297a4d660389d8d9
```

**Comment by ex-purple** [comment]
@Adela0814
> I got a bad hash error in my local

I cannot help with this. For me and Azure Pipelines, it does not reproduce.

**Comment by ex-purple** [comment]
@dg0yt 
Clipper and Clipper2 looks very difference:
https://sourceforge.net/projects/polyclipping/
https://sourceforge.net/p/polyclipping/code/HEAD/tree/tags/6.2.0/cpp/CMakeLists.txt

I don't think that we should change something in the typical Clipper2 installation

**Comment by dg0yt** [comment]
> @Adela0814
> 
> > I got a bad hash error in my local
> 
> I cannot help with this. For me and Azure Pipelines, it does not reproduce.

@ex-purple Maybe you local downloads folder (and vcpkg CI) has the asset with the SHA512 from GH yesterday's temporary change?

**Comment by dg0yt** [comment]
> @dg0yt Clipper and Clipper2 looks very difference: https://sourceforge.net/projects/polyclipping/ https://sourceforge.net/p/polyclipping/code/HEAD/tree/tags/6.2.0/cpp/CMakeLists.txt
> 
> I don't think that we should change something in the typical Clipper2 installation

I didn't ask for changes. I just wanted to avoid surprises if both libaries end up in the linking of a single executable.

The answer is : Clipper2 uses a different C++ namespace. So no really no problem.

File: ports/clipper2/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dg0yt) [inline ports/clipper2/portfile.cmake:2]: I see an unguarded 
~~~
#define EXTERN_DLL_EXPORT extern "C" __declspec(dllexport)
~~~
in `CPP/Clipper2Lib/include/clipper2/clipper.export.h`.

I guess for dynamic linkage support,  this must be replaced in the installed header?

**Comment by ex-purple** [comment]
> The answer is : Clipper2 uses a different C++ namespace. So no really no problem.

Thanks for the lesson. I'm not native speaker 🙂

> @ex-purple Maybe you local downloads folder (and vcpkg CI) has the asset with the SHA512 from GH yesterday's temporary change?

Maybe, Could you tell me what kind of change I should make?

File: ports/clipper2/portfile.cmake (hunk 2 above)
Comment (ex-purple) [reply ports/clipper2/portfile.cmake:2]: I have never experienced DDL export in practice, but I can try this option

**Comment by dg0yt** [comment]
> Could you tell me what kind of changes I should make?

Delete your local `downloads\AngusJohnson-Clipper2-Clipper2_1.1.1.tar.gz`, then rebuild. You will see that the hash changed.
(Yesterday a change of hashes at GH broke many existing recipes. Your recipe is broken because that change was rolled back. Cf. #29288)

File: ports/clipper2/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/clipper2/portfile.cmake:2]: We may change it in an update if you don't feel comfortable with it.

Comment (ex-purple) [reply ports/clipper2/portfile.cmake:2]: Yes, please 🙏

**Comment by ex-purple** [comment]
> Delete your local downloads\AngusJohnson-Clipper2-Clipper2_1.1.1.tar.gz, then rebuild. You will see that the hash changed.

Indeed
I updated actual hash

**Comment by BillyONeal** [comment]
I manually checked that the SHA in here is not affected by https://github.com/microsoft/vcpkg/issues/29288

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

@dg0yt Thanks for your review!

**Pull request merged**